bpf/lib/nodeport_egress.h

```diff
@@ -84,6 +84,24 @@ static __always_inline int nodeport_snat_fwd_ipv6(struct __ctx_buff *ctx,
 	if (IS_ERR(ret))
 		goto out;
 
+#if defined(ENABLE_EGRESS_GATEWAY_COMMON) && defined(IS_BPF_HOST)
+	if (target.egress_gateway) {
+		/* Stay on the desired egress interface: */
+		if (target.ifindex && target.ifindex == THIS_INTERFACE_IFINDEX)
+			goto apply_snat;
+
+		/* Send packet to the correct egress interface, and SNAT it there. */
+		ret = egress_gw_fib_lookup_and_redirect_v6(ctx, target.addr,
+							tuple.daddr, target.ifindex,
+							ext_err);
+		if (ret != CTX_ACT_OK)
```

Comment:
This is used during masquerading, for traffic by a client that's local on the gateway node. When it hits `to-netdev` (based on the default route) and the masquerading code sees that the desired `egressIP` lives on a **different** network interface, we bounce the packet over to the correct network interface (and then do the masquerading there).